lizmap/modules/lizmap/controllers/service.classic.php

```diff
@@ -1366,17 +1366,13 @@ public function DescribeFeatureType()
                     $go = false;
                 }
                 if ($go && $xml->complexType) {
-                    $layername = $layer->name;
-                    $typename = (string) $xml->complexType->attributes()->name;
-                    if ($typename == $layername.'Type') {
-                        $jsonData['name'] = $layername;
-                        $types = array();
-                        $elements = $xml->complexType->complexContent->extension->sequence->element;
-                        foreach ($elements as $element) {
-                            $types[(string) $element->attributes()->name] = (string) $element->attributes()->type;
-                        }
-                        $jsonData['types'] = (object) $types;
+                    $jsonData['name'] = $layer->name;
+                    $types = array();
+                    $elements = $xml->complexType->complexContent->extension->sequence->element;
```

Comment:
What if you check `$typename == $this->params['typename'].'Type'` ?

The describeFeatureType can retrieve multiple layers. We probably be sure that if we found `$layer` the parameter typename is for only one layer.